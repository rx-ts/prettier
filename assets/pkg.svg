<svg version="1.0" xmlns="http://www.w3.org/2000/svg" width="960" height="261" viewBox="120.5 116.5 960 261"><path opacity=".5" fill="#CFD4D7" d="M300.567 342h50a5 5 0 110 10h-50a5 5 0 110-10zm-80 0h60a5 5 0 110 10h-60a5 5 0 110-10z"/><path fill="#EA5E5E" d="M150.567 342h50a5 5 0 110 10h-50a5 5 0 110-10z"/><path opacity=".5" fill="#CFD4D7" d="M310.567 322h40a5 5 0 110 10h-40a5 5 0 110-10zm-30 0h10a5 5 0 110 10h-10a5 5 0 110-10zm-60 0h40a5 5 0 110 10h-40a5 5 0 110-10z"/><path fill="#F7B93E" d="M190.567 322h10a5 5 0 110 10h-10a5 5 0 110-10z"/><path fill="#56B3B4" d="M150.567 322h20a5 5 0 110 10h-20a5 5 0 110-10z"/><path opacity=".5" fill="#CFD4D7" d="M260.567 302h90a5 5 0 110 10h-90a5 5 0 110-10zm-40 0h20a5 5 0 110 10h-20a5 5 0 110-10z"/><path fill="#BF85BF" d="M150.567 302h50a5 5 0 110 10h-50a5 5 0 110-10z"/><path opacity=".5" fill="#CFD4D7" d="M330.567 282h20a5 5 0 110 10h-20a5 5 0 110-10zm-40 0h20a5 5 0 110 10h-20a5 5 0 110-10z"/><path fill="#F7B93E" d="M180.567 282h90a5 5 0 110 10h-90a5 5 0 110-10z"/><path fill="#BF85BF" d="M150.567 282h10a5 5 0 110 10h-10a5 5 0 110-10z"/><path opacity=".5" fill="#CFD4D7" d="M320.567 262h30a5 5 0 110 10h-30a5 5 0 110-10z"/><path fill="#BF85BF" d="M270.567 262h30a5 5 0 110 10h-30a5 5 0 110-10z"/><path fill="#EA5E5E" d="M210.567 262h40a5 5 0 110 10h-40a5 5 0 110-10z"/><path fill="#56B3B4" d="M150.567 262h40a5 5 0 110 10h-40a5 5 0 110-10z"/><path opacity=".5" fill="#CFD4D7" d="M330.567 242h20a5 5 0 110 10h-20a5 5 0 010-10z"/><path fill="#56B3B4" d="M230.567 242h80a5 5 0 110 10h-80a5 5 0 010-10z"/><path fill="#F7B93E" d="M180.567 242h30a5 5 0 110 10h-30a5 5 0 010-10z"/><path fill="#EA5E5E" d="M150.567 242h10a5 5 0 110 10h-10a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M340.567 222h10a5 5 0 010 10h-10a5 5 0 010-10z"/><path fill="#F7B93E" d="M270.567 222h50a5 5 0 010 10h-50a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M220.567 222h30a5 5 0 010 10h-30a5 5 0 010-10z"/><path fill="#BF85BF" d="M150.567 222h50a5 5 0 010 10h-50a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M340.567 202h10a5 5 0 010 10h-10a5 5 0 010-10z"/><path fill="#EA5E5E" d="M270.567 202h50a5 5 0 010 10h-50a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M220.567 202h30a5 5 0 010 10h-30a5 5 0 010-10z"/><path fill="#56B3B4" d="M190.567 202h10a5 5 0 010 10h-10a5 5 0 010-10z"/><path fill="#F7B93E" d="M150.567 202h20a5 5 0 010 10h-20a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M330.567 182h20a5 5 0 010 10h-20a5 5 0 010-10z"/><path fill="#56B3B4" d="M300.567 182h10a5 5 0 010 10h-10a5 5 0 010-10z"/><path fill="#BF85BF" d="M240.567 182h40a5 5 0 010 10h-40a5 5 0 010-10z"/><path fill="#56B3B4" d="M150.567 182h70a5 5 0 010 10h-70a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M320.567 162h30a5 5 0 010 10h-30a5 5 0 010-10z"/><path fill="#F7B93E" d="M200.567 162h100a5 5 0 010 10h-100a5 5 0 010-10z"/><path fill="#EA5E5E" d="M150.567 162h30a5 5 0 010 10h-30a5 5 0 010-10z"/><path opacity=".5" fill="#CFD4D7" d="M290.567 142h60a5 5 0 010 10h-60a5 5 0 010-10z"/><path fill="#56B3B4" d="M150.567 142h120a5 5 0 010 10h-120a5 5 0 010-10z"/><path d="M519.763 247.359c0 3.625-.512 6.756-1.534 9.397s-2.446 4.836-4.271 6.582c-1.825 1.748-3.98 3.043-6.466 3.883-2.485.842-5.19 1.264-8.116 1.264-.983 0-1.967-.066-2.951-.195a22.47 22.47 0 01-2.679-.504v15.416h-9.476v-54.871h8.193l.233 5.825a27.141 27.141 0 012.505-2.757c.867-.828 1.806-1.534 2.815-2.116s2.097-1.029 3.262-1.34 2.472-.466 3.922-.466c2.33 0 4.401.466 6.214 1.398 1.812.932 3.332 2.259 4.562 3.98 1.229 1.722 2.168 3.806 2.815 6.252.647 2.446.972 5.197.972 8.252zm-9.942.389c0-2.176-.155-4.026-.466-5.554-.311-1.527-.757-2.783-1.34-3.767-.582-.983-1.295-1.702-2.136-2.155-.842-.453-1.806-.68-2.893-.68-1.605 0-3.14.647-4.603 1.942s-3.009 3.055-4.64 5.281v17.241c.75.285 1.676.518 2.776.699 1.1.182 2.22.271 3.358.271 1.502 0 2.867-.311 4.098-.932 1.229-.621 2.277-1.502 3.146-2.641.866-1.139 1.533-2.529 1.999-4.174.469-1.642.701-3.486.701-5.531zm42.91 20.037l-.232-5.166c-.829.881-1.696 1.689-2.603 2.428s-1.902 1.379-2.99 1.922a15.9 15.9 0 01-3.553 1.262c-1.281.299-2.687.447-4.214.447-2.019 0-3.792-.297-5.319-.893-1.528-.596-2.81-1.424-3.845-2.486-1.036-1.061-1.819-2.342-2.35-3.844s-.796-3.158-.796-4.971c0-1.863.395-3.586 1.185-5.164.789-1.58 1.993-2.939 3.611-4.078 1.617-1.139 3.637-2.032 6.058-2.68 2.421-.646 5.262-.971 8.524-.971h5.164v-2.369c0-1.01-.143-1.922-.427-2.738a4.934 4.934 0 00-1.379-2.078c-.635-.569-1.456-1.009-2.466-1.32-1.01-.311-2.252-.466-3.728-.466-2.33 0-4.628.266-6.893.796a35.82 35.82 0 00-6.582 2.233v-7.572c1.889-.75 4.07-1.372 6.543-1.864a39.403 39.403 0 017.708-.738c2.926 0 5.437.278 7.533.834 2.098.557 3.825 1.392 5.185 2.505a9.953 9.953 0 013.01 4.175c.646 1.669.971 3.618.971 5.844v26.95h-8.115zm-1.359-17.435h-5.786c-1.605 0-2.964.154-4.077.465s-2.02.738-2.718 1.281c-.699.545-1.211 1.172-1.534 1.885a5.415 5.415 0 00-.485 2.271c0 1.605.518 2.828 1.553 3.67 1.035.84 2.447 1.262 4.233 1.262 1.32 0 2.692-.479 4.116-1.438 1.423-.957 2.99-2.33 4.698-4.115v-5.281zm52.27 15.92a29.46 29.46 0 01-5.786 1.67 34.859 34.859 0 01-6.252.545c-3.133 0-5.941-.428-8.427-1.281-2.485-.855-4.596-2.123-6.33-3.807-1.734-1.682-3.062-3.768-3.98-6.252s-1.379-5.359-1.379-8.621c0-3.158.499-6.019 1.495-8.582.997-2.563 2.401-4.75 4.214-6.563 1.812-1.812 3.993-3.21 6.543-4.194 2.55-.983 5.392-1.476 8.524-1.476 2.562 0 4.718.136 6.465.408 1.748.272 3.321.615 4.719 1.029v9.204a23.844 23.844 0 00-5.224-1.864c-1.877-.44-3.656-.66-5.339-.66-1.761 0-3.34.298-4.738.894-1.397.596-2.576 1.424-3.533 2.485-.958 1.062-1.689 2.35-2.194 3.864-.505 1.514-.757 3.178-.757 4.991 0 1.889.271 3.586.815 5.086.544 1.502 1.313 2.777 2.311 3.826a9.981 9.981 0 003.592 2.406c1.398.557 2.951.836 4.66.836.828 0 1.702-.059 2.621-.176.919-.115 1.838-.283 2.757-.504s1.825-.473 2.719-.758 1.729-.596 2.505-.932v8.426zm36.23 1.515l-14.562-19.727v19.727h-9.476v-54.871h9.476v32.115l13.785-16.698h11.844L634.94 246.31l17.319 21.475h-12.387zm44.036 0l-.232-5.166a27.03 27.03 0 01-2.602 2.428 15.927 15.927 0 01-6.544 3.184c-1.281.299-2.686.447-4.213.447-2.02 0-3.793-.297-5.32-.893-1.528-.596-2.81-1.424-3.845-2.486-1.035-1.061-1.818-2.342-2.349-3.844-.531-1.502-.797-3.158-.797-4.971 0-1.863.395-3.586 1.185-5.164.79-1.58 1.993-2.939 3.611-4.078s3.637-2.032 6.059-2.68c2.42-.646 5.262-.971 8.523-.971h5.165v-2.369c0-1.01-.143-1.922-.428-2.738s-.744-1.508-1.378-2.078c-.635-.569-1.457-1.009-2.467-1.32-1.009-.311-2.252-.466-3.728-.466-2.33 0-4.628.266-6.893.796a35.82 35.82 0 00-6.582 2.233v-7.572c1.89-.75 4.07-1.372 6.543-1.864a39.41 39.41 0 017.708-.738c2.926 0 5.437.278 7.534.834 2.097.557 3.825 1.392 5.184 2.505a9.953 9.953 0 013.01 4.175c.646 1.669.971 3.618.971 5.844v26.95h-8.115zm-1.358-17.435h-5.786c-1.605 0-2.965.154-4.078.465s-2.019.738-2.718 1.281c-.699.545-1.211 1.172-1.534 1.885a5.415 5.415 0 00-.485 2.271c0 1.605.518 2.828 1.554 3.67 1.035.84 2.446 1.262 4.232 1.262 1.32 0 2.692-.479 4.116-1.438 1.424-.957 2.99-2.33 4.699-4.115v-5.281zm49.938-15.107c.725.907 1.236 1.858 1.534 2.854.297.997.446 2.013.446 3.048 0 2.201-.383 4.142-1.146 5.825-.764 1.684-1.839 3.094-3.224 4.233s-3.042 2-4.97 2.582c-1.93.582-4.059.875-6.389.875-1.372 0-2.641-.137-3.806-.408s-2.045-.59-2.641-.951c-.44.439-.815.926-1.126 1.455-.311.531-.466 1.133-.466 1.807 0 .414.104.828.311 1.242s.511.783.913 1.107c.4.324.879.588 1.437.795a6.03 6.03 0 001.883.35l9.359.35c2.097.053 3.993.318 5.688.797s3.151 1.158 4.369 2.039c1.217.879 2.155 1.947 2.815 3.203s.99 2.686.99 4.291c0 1.838-.415 3.572-1.243 5.203-.829 1.631-2.077 3.049-3.747 4.252-1.67 1.205-3.747 2.156-6.232 2.855-2.485.697-5.372 1.047-8.66 1.047-3.185 0-5.922-.252-8.213-.756-2.291-.506-4.169-1.211-5.631-2.117-1.463-.906-2.543-1.986-3.242-3.242s-1.049-2.635-1.049-4.137c0-.906.116-1.746.35-2.523s.582-1.521 1.049-2.232a12.1 12.1 0 011.747-2.078 21.109 21.109 0 012.485-2.02c-1.269-.725-2.232-1.656-2.893-2.795s-.99-2.355-.99-3.65c0-.881.116-1.703.349-2.467a12.01 12.01 0 01.913-2.193c.375-.699.809-1.373 1.301-2.02a35.374 35.374 0 011.554-1.902c-.958-.957-1.755-2.117-2.389-3.476-.635-1.359-.951-3.061-.951-5.106 0-2.2.395-4.161 1.185-5.883.789-1.721 1.883-3.171 3.281-4.349 1.397-1.178 3.061-2.071 4.99-2.679 1.928-.608 4.031-.913 6.31-.913 1.165 0 2.285.058 3.359.175 1.074.116 2.052.317 2.932.602h13.669v6.912h-6.211zm-22.329 37.784c0 1.527.771 2.652 2.311 3.379 1.54.725 3.709 1.086 6.505 1.086 1.786 0 3.294-.168 4.523-.504 1.229-.338 2.233-.777 3.01-1.32s1.333-1.178 1.67-1.902a5.313 5.313 0 00.505-2.254c0-.697-.162-1.287-.485-1.766-.324-.48-.776-.881-1.359-1.205-.582-.322-1.269-.568-2.058-.736a16.98 16.98 0 00-2.621-.33l-8.35-.156c-.751.518-1.365 1.01-1.845 1.477-.479.465-.854.924-1.126 1.379a4.264 4.264 0 00-.544 1.377 7.689 7.689 0 00-.136 1.475zm1.593-31.649c0 2.097.614 3.761 1.844 4.99 1.229 1.229 2.945 1.844 5.146 1.844 1.165 0 2.181-.193 3.049-.582.867-.389 1.585-.906 2.155-1.554a6.472 6.472 0 001.281-2.252 8.434 8.434 0 00.427-2.679c0-2.2-.615-3.916-1.844-5.146-1.23-1.229-2.945-1.844-5.146-1.844-1.165 0-2.182.194-3.049.583-.867.388-1.586.913-2.155 1.573s-.997 1.431-1.281 2.311a8.89 8.89 0 00-.427 2.756zm69.238 3.651c0 .647-.02 1.489-.059 2.524a59.46 59.46 0 01-.175 2.912h-25.939c0 1.734.265 3.262.796 4.582.53 1.32 1.281 2.428 2.252 3.32.971.895 2.136 1.574 3.495 2.039 1.359.467 2.867.699 4.524.699 1.915 0 3.967-.148 6.154-.447a48.55 48.55 0 006.815-1.455v7.533c-1.01.285-2.11.543-3.301.777a78.77 78.77 0 01-3.65.621c-1.242.18-2.499.316-3.767.406a51.039 51.039 0 01-3.689.137c-3.003 0-5.695-.428-8.077-1.281s-4.401-2.117-6.058-3.787-2.926-3.732-3.806-6.193c-.881-2.459-1.32-5.307-1.32-8.543 0-3.209.439-6.129 1.32-8.757.88-2.627 2.122-4.879 3.728-6.757a16.38 16.38 0 015.845-4.349c2.291-1.022 4.848-1.534 7.669-1.534 2.797 0 5.269.434 7.418 1.301 2.148.868 3.947 2.078 5.397 3.631 1.449 1.553 2.55 3.404 3.301 5.553.751 2.149 1.127 4.505 1.127 7.068zm-9.631-1.36c.025-1.553-.168-2.906-.582-4.058s-.978-2.104-1.689-2.854a6.707 6.707 0 00-2.485-1.67 8.448 8.448 0 00-3.048-.543c-2.383 0-4.337.796-5.864 2.388s-2.434 3.838-2.718 6.737h16.386zm35.222 9.786c1.087 0 2.097.201 3.029.602.932.402 1.74.945 2.427 1.631.686.688 1.223 1.496 1.611 2.428s.583 1.93.583 2.99a7.442 7.442 0 01-.583 2.932 7.498 7.498 0 01-1.611 2.408 7.706 7.706 0 01-2.427 1.631c-.933.4-1.942.602-3.029.602a7.108 7.108 0 01-2.912-.602 7.556 7.556 0 01-3.961-4.039 7.439 7.439 0 01-.583-2.932c0-1.061.194-2.059.583-2.99a7.603 7.603 0 011.592-2.428 7.59 7.59 0 012.369-1.631 7.108 7.108 0 012.912-.602zm56.385 13.203c0 2.951-.434 5.521-1.301 7.709s-2.11 4-3.728 5.436c-1.618 1.438-3.561 2.512-5.825 3.225-2.266.711-4.796 1.066-7.592 1.066-2.278 0-4.401-.193-6.368-.582-1.968-.389-3.586-.854-4.854-1.398v-8.697c.725.389 1.547.77 2.466 1.145s1.877.713 2.874 1.01c.996.297 2.013.531 3.048.699s2.059.252 3.068.252c2.873 0 5.021-.828 6.445-2.484 1.424-1.658 2.137-4.092 2.137-7.301v-31.027h-18.757v-7.378h28.387v38.325zm.776-49.123c0 .854-.155 1.657-.466 2.408a5.96 5.96 0 01-1.301 1.961 6.25 6.25 0 01-4.446 1.805c-.88 0-1.695-.161-2.446-.485a6.215 6.215 0 01-1.961-1.32 5.945 5.945 0 01-1.301-1.961 6.224 6.224 0 01-.466-2.408 6.232 6.232 0 011.767-4.388 6.072 6.072 0 014.407-1.825c.88 0 1.702.162 2.466.486s1.424.771 1.98 1.339a6.203 6.203 0 011.767 4.388zm46.6 38.639c0 2.252-.499 4.174-1.495 5.766a11.825 11.825 0 01-3.961 3.885c-1.645.996-3.508 1.721-5.592 2.174s-4.213.68-6.388.68c-2.9 0-5.515-.137-7.845-.408s-4.53-.666-6.602-1.184v-8.543c2.434 1.01 4.86 1.74 7.281 2.193s4.705.68 6.854.68c2.485 0 4.342-.395 5.572-1.184 1.229-.791 1.845-1.818 1.845-3.088 0-.596-.13-1.139-.389-1.631s-.744-.965-1.456-1.418c-.713-.453-1.716-.918-3.01-1.398-1.295-.479-2.99-1.016-5.087-1.611-1.941-.543-3.645-1.158-5.106-1.844-1.463-.686-2.674-1.494-3.631-2.427a9.055 9.055 0 01-2.155-3.243c-.479-1.229-.719-2.672-.719-4.33 0-1.605.362-3.119 1.088-4.543.725-1.423 1.806-2.673 3.242-3.748 1.437-1.074 3.229-1.922 5.378-2.543 2.149-.621 4.66-.932 7.534-.932 2.485 0 4.686.13 6.602.388 1.915.259 3.611.543 5.087.854v7.728c-2.252-.725-4.369-1.236-6.35-1.534-1.98-.297-3.941-.446-5.883-.446s-3.502.35-4.68 1.048c-1.178.699-1.767 1.67-1.767 2.913 0 .596.116 1.126.35 1.592.233.466.692.919 1.379 1.359.686.44 1.644.9 2.873 1.378 1.229.479 2.854 1.004 4.874 1.573 2.277.647 4.193 1.34 5.747 2.078 1.553.738 2.802 1.573 3.747 2.505s1.624 1.994 2.039 3.184c.417 1.19.624 2.549.624 4.077zm46.716-8.426c0 3.105-.44 5.947-1.32 8.523-.881 2.576-2.149 4.783-3.806 6.621s-3.689 3.262-6.097 4.271-5.152 1.514-8.232 1.514c-2.926 0-5.54-.428-7.845-1.281-2.304-.854-4.259-2.129-5.863-3.824-1.605-1.695-2.835-3.818-3.689-6.369s-1.281-5.521-1.281-8.912c0-3.132.446-5.98 1.34-8.543s2.175-4.75 3.845-6.563c1.67-1.812 3.708-3.216 6.116-4.213 2.407-.996 5.125-1.495 8.154-1.495 2.951 0 5.579.434 7.883 1.301 2.305.868 4.259 2.155 5.864 3.864 1.604 1.708 2.828 3.825 3.67 6.349.84 2.523 1.261 5.443 1.261 8.757zm-9.864.232c0-4.091-.771-7.165-2.311-9.223-1.541-2.058-3.812-3.087-6.815-3.087-1.657 0-3.067.324-4.232.971s-2.123 1.534-2.874 2.66c-.751 1.126-1.301 2.44-1.65 3.941-.35 1.502-.524 3.107-.524 4.816 0 4.115.828 7.217 2.485 9.299 1.656 2.084 3.922 3.127 6.796 3.127 1.579 0 2.951-.316 4.116-.951s2.109-1.516 2.835-2.641c.725-1.127 1.268-2.453 1.631-3.98.362-1.527.543-3.172.543-4.932zm41.59 19.805v-25.708c0-4.323-1.605-6.485-4.815-6.485-1.605 0-3.14.647-4.602 1.942-1.463 1.294-3.01 3.055-4.641 5.281v24.97h-9.476V228.33h8.194l.232 5.825a27.378 27.378 0 012.505-2.757c.867-.828 1.806-1.534 2.815-2.116s2.097-1.029 3.262-1.34 2.472-.466 3.922-.466c2.02 0 3.78.33 5.281.99 1.502.66 2.758 1.592 3.768 2.796 1.009 1.204 1.767 2.654 2.271 4.349.505 1.696.757 3.592.757 5.689v26.484h-9.473zM477.974 308.6h-4.854c-8.438 0-14.713-1.979-18.822-5.936-4.11-3.958-6.165-9.959-6.165-18.003V262.1c0-2.142-.241-4.055-.721-5.738-.482-1.683-1.323-3.104-2.525-4.264-1.203-1.157-2.799-2.043-4.788-2.656-1.99-.611-4.493-.918-7.509-.918h-2.82v-9.116h2.82c3.191 0 5.792-.262 7.804-.787 2.011-.525 3.584-1.322 4.722-2.394 1.136-1.071 1.923-2.437 2.361-4.099.437-1.661.656-3.629.656-5.903v-15.543c0-3.673.437-6.985 1.312-9.936.874-2.951 2.306-5.465 4.295-7.542 1.989-2.076 4.568-3.673 7.739-4.788 3.17-1.115 7.05-1.672 11.641-1.672h4.854v9.182h-3.87c-10.144 0-15.215 4.919-15.215 14.756v15.281c0 10.668-4.613 16.681-13.838 18.035 9.313.919 13.97 6.909 13.97 17.971v22.429c0 10.012 5.027 15.02 15.084 15.02h3.87v9.182zm529.189-121.855h4.854c8.438 0 14.713 1.979 18.822 5.935 4.109 3.958 6.165 9.958 6.165 18.003v15.15c0 2.143.24 4.056.722 5.738.48 1.684 1.322 3.105 2.525 4.263 1.201 1.159 2.809 2.044 4.82 2.656 2.01.613 4.525.918 7.541.918h2.82v9.116h-2.82c-3.191 0-5.803.263-7.836.787s-3.619 1.323-4.756 2.394c-1.137 1.072-1.924 2.438-2.36 4.1-.438 1.662-.656 3.63-.656 5.902v22.954c0 3.673-.438 6.984-1.312 9.936-.875 2.952-2.295 5.466-4.263 7.543-1.968 2.076-4.548 3.672-7.739 4.787s-7.082 1.673-11.674 1.673h-4.854v-9.182h3.936c10.1 0 15.15-4.919 15.15-14.757V261.97c0-10.668 4.611-16.658 13.838-17.971-9.312-1.005-13.97-7.017-13.97-18.035v-15.019c0-10.012-5.007-15.019-15.019-15.019h-3.936v-9.181z" fill="#334650"/></svg>