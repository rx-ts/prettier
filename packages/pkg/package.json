{
  "name": "prettier-plugin-pkg",
  "version": "0.4.6",
  "description": "An opinionated package.json formatter plugin for Prettier",
  "repository": "git@github.com/rx-ts/prettier.git",
  "homepage": "https://github.com/rx-ts/prettier/blob/master/packages/pkg",
  "author": "JounQin <admin@1stg.me>",
  "license": "MPL-2.0",
  "main": "lib/cjs",
  "module": "lib",
  "es2015": "lib/es2015",
  "fesm5": "lib/esm",
  "jsdelivr": "lib/umd",
  "unpkg": "lib/umd",
  "types": "lib",
  "files": [
    "lib"
  ],
  "keywords": [
    "package",
    "package.json",
    "pkg",
    "plugin",
    "prettier",
    "prettier-plugin"
  ],
  "peerDependencies": {
    "prettier": "^1.19.1"
  }
}
