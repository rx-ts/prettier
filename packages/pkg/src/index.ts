import { ObjectProperty } from '@babel/types'
import { Plugin } from 'prettier'
import _ from 'prettier/parser-babylon'

import { engines } from './rules/engines'
import { files } from './rules/files'
import { scripts } from './rules/scripts'
import { sort } from './rules/sort'

const { 'json-stringify': parser } = _.parsers
const { parse } = parser

const format = (properties: ObjectProperty[]) => {
  let props = sort(properties)
  props = engines(props)
  props = files(props)
  props = scripts(props)
  return props
}

export default {
  name: 'prettier-plugin-pkg',
  parsers: {
    'json-stringify': {
      ...parser,
      parse(...args) {
        const [, , options] = args
        const { filepath } = options
        const ast = parse(...args)

        if (filepath.endsWith('package.json')) {
          const { properties } = ast
          ast.properties = format(properties)
        }

        return ast
      },
    },
  },
} as Plugin
