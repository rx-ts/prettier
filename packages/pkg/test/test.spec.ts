import { shuffle } from 'lodash'
import { join } from 'path'
import prettier, { Options } from 'prettier'
import PkgPlugin from 'prettier-plugin-pkg'

import pkg from './fixtures/fixture.json'

const createFixture = () =>
  shuffle(Object.keys(pkg)).reduce(
    (acc, key: keyof typeof pkg) =>
      Object.assign(acc, {
        [key]: pkg[key],
      }),
    {} as typeof pkg,
  )

test('randomize', () => {
  const input = JSON.stringify(createFixture(), null, 2)
  const output = prettier.format(input, {
    filepath: join(__dirname, 'package.json'),
    parser: 'json-stringify',
    plugins: [PkgPlugin],
  })

  expect(output).toMatchSnapshot()
})

test('preprocess', () => {
  const input = JSON.stringify(createFixture(), null, 2)
  const output = prettier.format(input, {
    filepath: join('package.json'),
    parser: 'json-stringify',
    plugins: [PkgPlugin],
    preprocess(content: string) {
      const { version, repository }: typeof pkg = JSON.parse(content)
      const result = { repository, version }
      return result
    },
  } as Options)

  expect(output).toMatchSnapshot()
})

test('not package.json', () => {
  const fixture = { version: 'batman', name: 'joker' }
  const input = JSON.stringify(fixture, null, 2)
  const output = prettier.format(input, {
    filepath: 'batman.json',
    parser: 'json-stringify',
    plugins: [PkgPlugin],
  })

  expect(input.trim()).toBe(output.trim())
})

test('broken json', () => {
  const broken = `{
  "name": "prettier-plugin-pkg",
  "batman": {]
}`

  expect(() =>
    prettier.format(broken, {
      filepath: 'broken.json',
      parser: 'json-stringify',
      plugins: [PkgPlugin],
    }),
  ).toThrow()
})
