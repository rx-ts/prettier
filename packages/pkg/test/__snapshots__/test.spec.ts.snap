// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preprocess 1`] = `
"{
  \\"name\\": \\"@rxts/prettier-plugin-pkg\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"An opinionated package.json formatter plugin for Prettier\\",
  \\"license\\": \\"MPL-2.0\\",
  \\"repository\\": \\"git@github.com/rx-ts/prettier.git\\",
  \\"author\\": \\"JounQin <admin@1stg.me>\\",
  \\"homepage\\": \\"https://github.com/rx-ts/prettier/blob/master/packages/pkg\\",
  \\"main\\": \\"lib/cjs\\",
  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\"
  },
  \\"files\\": [
    \\"lib\\",
    \\"rules\\"
  ],
  \\"keywords\\": [
    \\"package\\",
    \\"package.json\\",
    \\"pkg\\",
    \\"plugin\\",
    \\"prettier\\",
    \\"prettier-plugin\\"
  ],
  \\"peerDependencies\\": {
    \\"prettier\\": \\"^1.18.2\\"
  },
  \\"es2015\\": \\"lib/es2015\\",
  \\"module\\": \\"lib/esm\\"
}
"
`;

exports[`randomize 1`] = `
"{
  \\"name\\": \\"@rxts/prettier-plugin-pkg\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"An opinionated package.json formatter plugin for Prettier\\",
  \\"license\\": \\"MPL-2.0\\",
  \\"repository\\": \\"git@github.com/rx-ts/prettier.git\\",
  \\"author\\": \\"JounQin <admin@1stg.me>\\",
  \\"homepage\\": \\"https://github.com/rx-ts/prettier/blob/master/packages/pkg\\",
  \\"main\\": \\"lib/cjs\\",
  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\"
  },
  \\"files\\": [
    \\"lib\\",
    \\"rules\\"
  ],
  \\"keywords\\": [
    \\"package\\",
    \\"package.json\\",
    \\"pkg\\",
    \\"plugin\\",
    \\"prettier\\",
    \\"prettier-plugin\\"
  ],
  \\"peerDependencies\\": {
    \\"prettier\\": \\"^1.18.2\\"
  },
  \\"es2015\\": \\"lib/es2015\\",
  \\"module\\": \\"lib/esm\\"
}
"
`;
